src/linear_algebra/tensor_product.lean

```diff
@@ -608,19 +608,117 @@ rfl
 
 end tensor_product
 
+namespace linear_map
+
+variables {R} (M) {N P Q}
+
+/-- `ltensor M f : M ⊗ N →ₗ M ⊗ P` is the natural linear map induced by `f : N →ₗ P`. -/
+def ltensor (f : N →ₗ[R] P) : M ⊗ N →ₗ[R] M ⊗ P :=
+tensor_product.map id f
+
+/-- `rtensor f M : N₁ ⊗ M →ₗ N₂ ⊗ M` is the natural linear map induced by `f : N₁ →ₗ N₂`. -/
+def rtensor (f : N →ₗ[R] P) : N ⊗ M →ₗ[R] P ⊗ M :=
+tensor_product.map f id
+
+variables (g : P →ₗ[R] Q) (f : N →ₗ[R] P)
+
+@[simp] lemma ltensor_tmul (m : M) (n : N) : f.ltensor M (m ⊗ₜ n) = m ⊗ₜ (f n) := rfl
+
+@[simp] lemma rtensor_tmul (m : M) (n : N) : f.rtensor M (n ⊗ₜ m) = (f n) ⊗ₜ m := rfl
+
+open tensor_product
+
+/-- `ltensor_hom M` is the natural linear map that sends a linear map `f : N →ₗ P` to `M ⊗ f`. -/
+def ltensor_hom : (N →ₗ[R] P) →ₗ[R] (M ⊗[R] N →ₗ[R] M ⊗[R] P) :=
```

Comment:
I don't know if it actually simplifies this PR at all, but is there / should there be a `tensor_producr.map_hom` that gives the bilinear map?

Reply:
I think we should redefine `tensor_product.map` to be that bundled hom. And `linear_map.tensor` can enable the notation `f.tensor g`.

Reply:
Can we do anything to make that notation `f ⊗ₜ g` instead? (And is that a good idea?)

Reply:
If we work in the category `Module R`, then that notation already works. It would be good to make it as easy as possible to shift back and forth between the two points of view.